Library/OcBootManagementLib/BootArguments.c

```diff
@@ -21,10 +12,33 @@
 #include <Library/OcMiscLib.h>
 #include <Library/PrintLib.h>
 #include <Library/UefiLib.h>
+#include <Library/UefiBootServicesTableLib.h>
 #include <Library/UefiRuntimeServicesTableLib.h>
 
 #include "BootManagementInternal.h"
 
+/*
+  LoadOptions processing states.
+*/
+typedef enum LOAD_OPTIONS_PARSE_STATE_ {
+  LOAD_OPTIONS_WHITE_SPACE,
+  LOAD_OPTIONS_NAME,
+  LOAD_OPTIONS_VALUE_FIRST,
+  LOAD_OPTIONS_VALUE,
+  LOAD_OPTIONS_QUOTED_VALUE
+} LOAD_OPTIONS_PARSE_STATE;
+
+#define IS_SPACE(c) ((c) == L' ' || (c) == L'\t' || (c) == L'\n')
```

Comment:
Better to put to OcStringLib and perhaps make a function.

Reply:
I guess I didn't want to tightly couple different programs which might have slightly different requirements for what counts as white space - but can do...

Reply:
Hmm, better to follow ctype here really. It matches a few more symbols, but I believe we can live with it.